src/lib/components/Notifier.svelte

```diff
@@ -0,0 +1,54 @@
+<script lang="ts">
+    import { onMount, tick } from "svelte";
+
+    let showNotice = false
+    let permissionGranted = false
+
+    async function requestPermission() {
+        showNotice = true
+        await tick() // show the notice before we requrest the permission
+        const permission = await Notification.requestPermission()
+        showNotice = false
+        return permission
+    }
+    
+    onMount(async () => {
+        if (!('Notification' in window)) {
+            console.log("This browser does not support the Notification API")
+        } else {
+            try {
+                if (Notification.permission === 'default') {
```

Comment:
I think it would be less intrusive if it was:
1. show that the site supports notification (with the "If you enable t..." block)
2. have a "enable" button below that requests for permission
3. the user can enable it anytime in the customize menu

Reply:
so this dialog only displays while the browser shows the "allow notifications?" popup. not anytime else. 
i've achieved this by first checking if the state is default. it if is, i show this, await tick, then ask for permission, and hide immediately after, regardless of the result.

the enabling/disabling is handled by the browser like this:
![image](https://user-images.githubusercontent.com/21956756/185250593-b5ea1b1d-9232-4f55-9cdd-c52d6e2eccc0.png)  
you don't need to add a custom enable button. 
  
i don't think it really makes sense for the user to see this as an option in customize since once the browser decides, the site can't change it, only the user from the browser, here:  
![image](https://user-images.githubusercontent.com/21956756/185250856-fc539291-fd9d-4e1b-b75c-4d2d89895ba2.png)

Reply:
It's fine on Chrome and Firefox but Safari forces you to pick one, so it's better to ask for permission after user interaction

<img width="1440" alt="Screen Shot 2022-08-18 at 7 11 55" src="https://user-images.githubusercontent.com/80624252/185252593-3ec40e49-1a6f-4033-a99c-23ed732c6a03.png">

Reply:
![image](https://user-images.githubusercontent.com/21956756/185258280-418c8841-515b-4747-a11a-b49670df20c5.png)
![image](https://user-images.githubusercontent.com/21956756/185258291-6dd5e922-dc36-46c0-94c6-9d368a397905.png)
changed it to this

Reply:
looks good to me